addons/website_sale/views/snippets.xml

```diff
@@ -61,16 +61,22 @@
     </xpath>
 </template>
 
+<template id="website_sale_search_box" inherit_id="website.website_search_box">
+    <xpath expr="//input[@name='search']" position="attributes">
+        <attribute name="class">search-query form-control o_wsale_search_products oe_search_box</attribute>
```

Comment:
Use add=" .." separator="..." like you already did before instead of overriding all the classes